src/sentry/preprod/vcs/status_checks/templates.py

```diff
@@ -1,138 +1,209 @@
 from __future__ import annotations
 
-from dataclasses import dataclass
+from enum import Enum
 
 from django.utils.translation import gettext_lazy as _
+from django.utils.translation import ngettext
 
-from sentry.preprod.models import PreprodArtifact
+from sentry.preprod.models import PreprodArtifact, PreprodArtifactSizeMetrics
 
 
-@dataclass(frozen=True)
-class _StatusCheckData:
-    """Data for formatting status check messages."""
+class _OverallStatus(Enum):
+    PROCESSING = "processing"
+    FAILED = "failed"
+    SUCCESS = "success"
 
-    build_id: int
-    app_id: str
-    version: str
 
-
-@dataclass(frozen=True)
-class _SuccessStatusCheckData(_StatusCheckData):
-    """Extended data for successful status checks with size metrics."""
-
-    download_size: str
-    download_change: str
-    install_size: str
-    install_change: str
-
-
-@dataclass(frozen=True)
-class _FailureStatusCheckData(_StatusCheckData):
-    """Extended data for failed status checks with error information."""
-
-    error_message: str
+_SIZE_ANALYZER_TITLE_BASE = _("Size Analysis")
 
 
 def format_status_check_messages(
-    preprod_artifact: PreprodArtifact,
+    artifacts: list[PreprodArtifact],
+    size_metrics_map: dict[int, PreprodArtifactSizeMetrics],
```

Comment:
I think this will need to be `dict[int, list[PreprodArtifactSizeMetrics]]` since an artifact can have multiple `PreprodArtifactSizeMetrics`.

We can temporarily hardcode to just be the `MAIN_ARTIFACT` type, but I think it would be best for us to just support the multiple artifacts from the get-go here to avoid broken states in the future